### DISCUSSION TITLE
configuration over ENV

### DISCUSSION THREAD

**Discussion by Borzo-terrum** [GreptimeTeam/greptimedb#4793] [Q&A] [answered]
Hi guys,

I use 
greptimedb-operator - helm version 0.2.5
greptimedb-cluster - helm version 0.2.14

According to the documentation, I can set the settings using environment variables

Trying to set region_engine.mito.page_cache_size

GREPTIMEDB_DATANODE__REGION_ENGINE__MITO__PAGE_CACHE_SIZE

I get an error 
datanode Error: 0: Failed to load layered config, at src/cmd/src/datanode.rs:152:10
datanode 1: Failed to load layered config, at /greptimedb/src/common/config/src/config.rs:79:14
datanode 2: invalid type: map, expected a sequence

Translated with www.DeepL.com/Translator (free version)

**Comment by zyy17** [comment, marked as answer]
We use [config-rs](https://github.com/mehcode/config-rs) to parse environment variables to config struct. Unfortunately, it [doesn't support](https://github.com/mehcode/config-rs/issues/536) parsing struct list(The type of [`region_engine`](https://github.com/GreptimeTeam/greptimedb/blob/main/config/datanode.example.toml#L388-L391) is list) from environment variable.

If you want to configure `region_engine`, you can use [`datanode.configData`](https://github.com/GreptimeTeam/helm-charts/blob/main/charts/greptimedb-cluster/values.yaml#L233-L241) of `greptimedb-cluster` chart. 

For example, you can create a [`values.yaml`](https://github.com/GreptimeTeam/helm-charts/tree/main/charts/greptimedb-cluster):

```yaml
datanode:
  configData: |-
    [[region_engine]]
    [region_engine.mito]
    page_cache_size = "256MB"
```

Upgrade your cluster by the following command:

```console
helm upgrade --install mycluster greptime/greptimedb-cluster --values ./values.yaml
```

We are considering extracting commonly used configurations to CRD fields that the user can configure easily without specifying environment variables or config data.

**Reply by Borzo-terrum** [reply]
thanks
